reformat_gherkin/core.py

```diff
@@ -39,23 +41,48 @@ def find_sources(src: Tuple[str]) -> Set[Path]:
 
 
 def reformat(src: Tuple[str], report: Report, *, options: Options):
-    sources = find_sources(src)
+    use_stdin = "-" in src
+    sources = find_sources(filter((lambda it: it != "-"), src))
 
-    if not sources:
+    if not sources and not use_stdin:
         raise EmptySources
 
+    if use_stdin:
+        changed = reformat_stdin(options=options)
+        report.done("stdin", changed)
+
     for path in sources:
         try:
             changed = reformat_single_file(path, options=options)
-            report.done(path, changed)
+            report.done(str(path), changed)
         except Exception as e:
             report.failed(path, str(e))
 
 
-# noinspection PyTypeChecker
+def reformat_stdin(*, options: Options) -> bool:
+    output = sys.stdout.buffer if options.write_back == WriteBackMode.INPLACE else None
+    return reformat_streams(sys.stdin.buffer, output, options=options)
+
+
 def reformat_single_file(path: Path, *, options: Options) -> bool:
-    with open(path, "rb") as buf:
-        src_contents, encoding, existing_newline = decode_bytes(buf.read())
+    with open(path, "rb") as f:
+        in_stream = BytesIO(f.read())
+
+    out_stream = (
+        open(path, "wb") if options.write_back == WriteBackMode.INPLACE else None
+    )
+
+    try:
+        return reformat_streams(in_stream, out_stream, options=options)
+    finally:
+        if out_stream is not None:
+            out_stream.close()
+
+
+def reformat_streams(
+    input: BinaryIO, output: Optional[BinaryIO], *, options: Options
```

Comment:
`input` is a built-in function, maybe we should have a different name?

Reply:
Ah, whoops!